lib/ansible/modules/monitoring/graylog_index_sets.py

```diff
@@ -0,0 +1,405 @@
+            retention_strategy_class=dict(type='str', default='org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy'),
+            rotation_strategy=dict(type='dict', default=dict(type='org.graylog2.indexer.rotation.strategies.TimeBasedRotationStrategyConfig',
+                                   rotation_period='P1D')),
+            retention_strategy=dict(type='dict', default=dict(type='org.graylog2.indexer.retention.strategies.DeletionRetentionStrategyConfig',
+                                    max_number_of_indices=14)),
+            index_optimization_max_num_segments=dict(type='int', default=1),
+            index_optimization_disabled=dict(type='bool', default=False),
+            writable=dict(type='bool', default=True),
+            default=dict(type='bool', default=False)
+        )
+    )
+
+    endpoint = module.params['endpoint']
+    graylog_user = module.params['graylog_user']
+    graylog_password = module.params['graylog_password']
+    action = module.params['action']
+    title = module.params['title']
+    index_set_id = module.params['index_set_id']
+    description = module.params['description']
+    index_prefix = module.params['index_prefix']
+    index_analyzer = module.params['index_analyzer']
+    shards = module.params['shards']
+    replicas = module.params['replicas']
+    rotation_strategy_class = module.params['rotation_strategy_class']
+    retention_strategy_class = module.params['retention_strategy_class']
+    rotation_strategy = module.params['rotation_strategy']
+    retention_strategy = module.params['retention_strategy']
+    index_optimization_max_num_segments = module.params['index_optimization_max_num_segments']
+    index_optimization_disabled = module.params['index_optimization_disabled']
+    writable = module.params['writable']
+    default = module.params['default']
+    creation_date = datetime.datetime.utcnow().isoformat() + 'Z'
+
+    base_url = "https://%s/api/system/indices/index_sets" % (endpoint)
+
+    api_token = get_token(module, endpoint, graylog_user, graylog_password)
+    headers = '{ "Content-Type": "application/json", "X-Requested-By": "Graylog API", "Accept": "application/json", \
+                "Authorization": "Basic ' + api_token.decode() + '" }'
+
+    if action == "create":
+        status, message, content, url = create(module, base_url, headers)
+    elif action == "update":
+        status, message, content, url = update(module, base_url, headers)
+    elif action == "delete":
+        status, message, content, url = delete(module, base_url, headers, index_set_id)
+    elif action == "list":
+        status, message, content, url = list(module, base_url, headers, index_set_id)
+    elif action == "query_index_sets":
+        index_set_id = query_index_sets(module, base_url, headers, title)
+        status, message, content, url = list(module, base_url, headers, index_set_id)
+
+    uresp = {}
+    content = to_text(content, encoding='UTF-8')
```

Comment:
This is not required since we are already doing this in individual APIs.

Reply:
are you referring to the `content = to_text(content, encoding='UTF-8')` ?